angular/projects/global/src/lib/components/table-template/table-template-utils.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable } from '@angular/core';
+import { Router, Params } from '@angular/router';
+import { PlatformLocation } from '@angular/common';
+import {
+  TableObject,
+  DEFAULT_TABLE_PAGE_SIZE,
+  DEFAULT_TABLE_CURRENT_PAGE,
+  DEFAULT_TABLE_SORT_BY,
+  DEFAULT_TABLE_KEYWORDS
+} from './table-object';
+
+@Injectable()
+export class TableTemplateUtils {
+  constructor(private platformLocation: PlatformLocation, private router: Router) {}
+
+  public updateUrl(tableObject: TableObject) {
+    if (tableObject == null) {
+      throw Error('Navigation Object cannot be null.');
+    }
+
+    let currentUrl = this.getBaseUrl();
+    this.setPaginationInUrl(currentUrl, tableObject.currentPage, tableObject.pageSize);
+    this.setKeywordsInUrl(currentUrl, tableObject.keywords);
+    this.setSortByInUrl(currentUrl, tableObject.sortBy);
+    this.setFilterInUrl(currentUrl, tableObject.filter);
+    currentUrl += ';ms=' + new Date().getTime();
+    window.history.replaceState({}, '', currentUrl);
+  }
+
+  public getBaseUrl(): string {
+    let currentUrl = this.router.url;
+    currentUrl = (this.platformLocation as any).getBaseHrefFromDOM() + currentUrl.slice(1);
+    currentUrl = currentUrl.split(';')[0];
+    return currentUrl;
+  }
+
+  public setPaginationInUrl(currentUrl: string, currentPage: number, pageSize: number) {
+    currentUrl += `;currentPage=${currentPage ? currentPage : DEFAULT_TABLE_CURRENT_PAGE};pageSize=${
```

Comment:
Instead of using a ternary:``` currentPage ? currentPage : DEFAULT_TABLE_CURRENT_PAGE ```
I would use the short circuit style: ``` currentPage || DEFAULT_TABLE_CURRENT_PAGE ```